pkg/store/kotsstore/downstream_store.go

```diff
@@ -359,24 +359,33 @@ func (s *KOTSStore) FindDownstreamVersions(appID string, downloadedOnly bool) (*
 	return nil, errors.New("app has no versions")
 }
 
-func (s *KOTSStore) GetDownstreamVersionHistory(appID string, clusterID string, currentPage int, pageSize int, pinLatest bool) ([]*downstreamtypes.DownstreamVersion, error) {
+func (s *KOTSStore) GetDownstreamVersionHistory(appID string, clusterID string, currentPage int, pageSize int, pinLatest bool, pinLatestDeployable bool) ([]*downstreamtypes.DownstreamVersion, error) {
 	versions, err := s.GetDownstreamVersions(appID, clusterID, false)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get downstream versions without details")
 	}
 
-	startIndex := currentPage * pageSize
-	endIndex := currentPage*pageSize + pageSize
+	desiredVersions := []*downstreamtypes.DownstreamVersion{}
+
 	if pinLatest {
-		startIndex++
-		endIndex++
+		if len(versions.AllVersions) > 0 {
+			desiredVersions = append(desiredVersions, versions.AllVersions[0])
+		}
 	}
-	desiredVersions := []*downstreamtypes.DownstreamVersion{}
-	for i, v := range versions.AllVersions {
-		if pinLatest && i == 0 {
-			desiredVersions = append(desiredVersions, v)
-			continue
+
+	if pinLatestDeployable {
```

Comment:
so latest and latest deployable will always be included as first for ALL pages and page size will always be desired page size + 2 if both flags are enabled. is this intended?

Reply:
yes